src/react/createConnectedStoreAs.tsx

```diff
@@ -65,37 +66,36 @@ export function createConnectedStoreAs<
         fx(stores as any) // TODO
       }
 
+      this.storeDefinitions = stores as any
+      this.subscriptions = this._createSubscriptions()
       this.state = {
-        storeDefinitions: stores as any, // TODO
         // TODO
-        storeSnapshots: mapValues(stores, _ => _.getCurrentSnapshot()) as any,
-        subscriptions: mapValues(stores, (_, k) =>
-          _.onAll().subscribe(() =>
-            this.setState(state => ({
-              storeSnapshots: Object.assign({}, state.storeSnapshots, {
-                [k]: _.getCurrentSnapshot()
-              })
-            }))
-          )
+        storeSnapshots: mapValues(stores, _ => _.getCurrentSnapshot()) as any
+      }
+    }
+
+    _createSubscriptions = () =>
+      mapValues(this.storeDefinitions, (_, k) =>
+        _.onAll().subscribe(() =>
+          this.setState(state => ({
+            storeSnapshots: Object.assign({}, state.storeSnapshots, {
+              [k]: _.getCurrentSnapshot()
+            })
+          }))
         )
+      ) as any
+
+    componentDidMount(): void {
+      if (this.subscriptions == null) {
+        this.subscriptions = this._createSubscriptions()
       }
     }
 
     componentWillUnmount() {
-      mapValues(this.state.subscriptions, _ => _.unsubscribe())
-      // Let the state get GC'd.
-      // TODO: Find a more elegant way to do this.
-      if (this.state.storeSnapshots) {
+      if (this.subscriptions != null) {
+        mapValues(this.subscriptions, _ => _.unsubscribe())
+        this.subscriptions = null
       }
-      mapValues(this.state.storeSnapshots, _ => ((_ as any).state = null))
-      mapValues(
-        this.state.storeSnapshots,
-        _ => ((_ as any).storeDefinition = null)
```

Comment:
Same here -- is there a way we could keep this GC-related code?

Reply:
same as above, we need to re-establish the subscriptions.